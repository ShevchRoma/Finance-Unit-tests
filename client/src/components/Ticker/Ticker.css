.bottom__column {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 50px;
}
.bottom__column > * {
  flex: 20%;
}
.bottom__column:last-child {
  margin-bottom: 0;
}
.bottom__name {
  display: flex;
  flex: 70%;
  align-items: center;
}
.bottom__icon {
  padding: 4px 8px;
  margin-right: 14px;
  background: rgb(99, 98, 98);
  border-radius: 5px;
  width: 60px;
  display: flex;
  justify-content: center;
}
.bottom__icon:last-child {
  background: red;
}
.bottom__icon > span {
  color: #fff;
  font-size: 14px;
  font-weight: 700;
}
.bottom__title {
  font-weight: bold;
}
.bottom__price {
  display: flex;
}
.bottom__rate {
  display: flex;
}
