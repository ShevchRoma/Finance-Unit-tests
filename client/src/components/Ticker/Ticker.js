import React from "react";
import "./Ticker.css";

const Ticker = ({ item }) => {
  return (
    <div className="bottom__column">
      <div className="bottom__name">
        <div className="bottom__icon">
          <span>{item?.ticker}</span>
        </div>
        <div className="bottom__title">{item?.ticker}</div>
      </div>
      <div className="bottom__price">
        <p>{item?.price}</p>
        <span>$</span>
      </div>
      <div className="bottom__rate">
        <p>{item?.change}</p>
        <span>$</span>
      </div>
      <div className="bottom__change">
        <p>{item?.change_percent}%</p>
      </div>
    </div>
  );
};

export default Ticker;
