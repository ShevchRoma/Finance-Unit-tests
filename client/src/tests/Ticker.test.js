import { cleanup } from "@testing-library/react";
import Ticker from "../components/Ticker/Ticker";
import TestRenderer from "react-test-renderer";

afterEach(cleanup);

describe("Ticker.js", () => {
  it("snapshot renders correctly, truthy values", () => {
    const tree = TestRenderer.create(
      <Ticker
        ticker="AAPL"
        price="270.51"
        change="110.94"
        change_percent="0.29"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("snapshot renders correctly, falsy values", () => {
    const tree = TestRenderer.create(<Ticker />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
