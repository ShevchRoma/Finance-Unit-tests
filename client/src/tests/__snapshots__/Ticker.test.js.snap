// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticker.js snapshot renders correctly, falsy values 1`] = `
<div
  className="bottom__column"
>
  <div
    className="bottom__name"
  >
    <div
      className="bottom__icon"
    >
      <span />
    </div>
    <div
      className="bottom__title"
    />
  </div>
  <div
    className="bottom__price"
  >
    <p />
    <span>
      $
    </span>
  </div>
  <div
    className="bottom__rate"
  >
    <p />
    <span>
      $
    </span>
  </div>
  <div
    className="bottom__change"
  >
    <p>
      %
    </p>
  </div>
</div>
`;

exports[`Ticker.js snapshot renders correctly, truthy values 1`] = `
<div
  className="bottom__column"
>
  <div
    className="bottom__name"
  >
    <div
      className="bottom__icon"
    >
      <span />
    </div>
    <div
      className="bottom__title"
    />
  </div>
  <div
    className="bottom__price"
  >
    <p />
    <span>
      $
    </span>
  </div>
  <div
    className="bottom__rate"
  >
    <p />
    <span>
      $
    </span>
  </div>
  <div
    className="bottom__change"
  >
    <p>
      %
    </p>
  </div>
</div>
`;
